clippy_lints/src/assign_ops.rs

```diff
@@ -50,7 +50,7 @@ declare_clippy_lint! {
     /// ### Known problems
     /// Clippy cannot know for sure if `a op= a op b` should have
     /// been `a = a op a op b` or `a = a op b`/`a op= b`. Therefore, it suggests both.
-    /// If `a op= a op b` is really the correct behaviour it should be
+    /// If `a op= a op b` is really the correct behavior it should be
```

Comment:
nits: They seems to be the same word.

Reply:
Although they're indeed the same word, "behavior" was used in more places than in others, so unified the expression to "behavior".

```
$ git grep behavior
CHANGELOG.md:* [`manual_split_once`]: No longer suggests code changing the original behavior
CODE_OF_CONDUCT.md:* We will exclude you from interaction if you insult, demean or harass anyone. That is not welcome behavior. We
CODE_OF_CONDUCT.md:  definition. In particular, we don't tolerate behavior that excludes people in socially marginalized groups.
CODE_OF_CONDUCT.md:* Likewise any spamming, trolling, flaming, baiting or other attention-stealing behavior is not welcome.
clippy_lints/src/booleans.rs:    /// Ignores short circuiting behavior of `||` and
clippy_lints/src/booleans.rs:    /// Ignores short circuiting behavior.
clippy_lints/src/booleans.rs:                // QMC has exponentially slow behavior as the number of terminals increases
clippy_lints/src/casts/cast_ref_to_mut.rs:                "casting `&T` to `&mut T` may cause undefined behavior, consider instead using an `UnsafeCell`",
clippy_lints/src/casts/mod.rs:    /// truncate large values. This is expected behavior, so the cast is `Allow` by
clippy_lints/src/casts/mod.rs:    /// be surprising when this is not the intended behavior, as demonstrated by the
clippy_lints/src/casts/mod.rs:    /// behavior.
clippy_lints/src/copies.rs:    ///   the if condition. The suggestion can in that case modify the behavior of the program.
clippy_lints/src/default_numeric_fallback.rs:    /// can be a pitfall that cause unexpected runtime behavior.
clippy_lints/src/default_union_representation.rs:    /// specified layout. These cases may lead to undefined behavior in unsafe blocks.
clippy_lints/src/deprecated_lints.rs:    /// between non-pointer types of differing alignment is well-defined behavior (it's semantically
clippy_lints/src/deprecated_lints.rs:    /// counterparts, so this lint may suggest a change in behavior or the code may not compile.
clippy_lints/src/deprecated_lints.rs:    "the replacement suggested by this lint had substantially different behavior"
clippy_lints/src/lib.deprecated.rs:        "the replacement suggested by this lint had substantially different behavior",
clippy_lints/src/lib.rs:        "the replacement suggested by this lint had substantially different behavior",
clippy_lints/src/manual_ok_or.rs:    /// Concise code helps focusing on behavior instead of boilerplate.
clippy_lints/src/manual_unwrap_or.rs:    /// Concise code helps focusing on behavior instead of boilerplate.
clippy_lints/src/mem_replace.rs:    /// This will lead to undefined behavior even if the
clippy_lints/src/mem_replace.rs:    ///     let new_v = may_panic(taken_v); // undefined behavior on panic
clippy_lints/src/methods/mod.rs:    /// For most types, this is undefined behavior.
clippy_lints/src/methods/uninit_assumed_init.rs:                "this call for this type may be undefined behavior"
clippy_lints/src/mutex_atomic.rs:                         behavior and not the internal type, consider using `Mutex<()>`",
clippy_lints/src/non_copy_const.rs:            // case (that actually removes another suboptimal behavior (I won't say 'false positive') where,
clippy_lints/src/ptr.rs:    /// This causes undefined behavior.
clippy_lints/src/ptr.rs:    /// // Bad. Undefined behavior
clippy_lints/src/transmute/mod.rs:    /// This is undefined behavior.
clippy_lints/src/transmuting_null.rs:    /// Transmuting a null pointer is undefined behavior.
clippy_lints/src/transmuting_null.rs:    "transmutes from a null pointer to a reference, which is undefined behavior"
clippy_lints/src/uninit_vec.rs:    /// undefined behavior with most safe operations. Notably, uninitialized
clippy_lints/src/uninit_vec.rs:    /// reader.read(&mut vec); // undefined behavior!
clippy_lints/src/wildcard_imports.rs:    /// This can lead to confusing error messages at best and to unexpected behavior at worst.
clippy_utils/src/qualify_min_const_fn.rs:// of terminologies might not be relevant in the context of Clippy. Note that its behavior might
doc/adding_lints.md:compile-test tests run on the 2015 edition by default. To change this behavior
doc/adding_lints.md:behavior that can be seen as a false positive for some users. Adding a configuration is done
tests/ui/auxiliary/option_helpers.rs:/// Utility macro to test linting behavior in `option_methods()`
tests/ui/borrow_interior_mutable_const/enums.rs:        // This is the "suboptimal behavior" mentioned in `is_value_unfrozen`
tests/ui/cast_ref_to_mut.stderr:error: casting `&T` to `&mut T` may cause undefined behavior, consider instead using an `UnsafeCell`
tests/ui/cast_ref_to_mut.stderr:error: casting `&T` to `&mut T` may cause undefined behavior, consider instead using an `UnsafeCell`
tests/ui/cast_ref_to_mut.stderr:error: casting `&T` to `&mut T` may cause undefined behavior, consider instead using an `UnsafeCell`
tests/ui/cognitive_complexity.rs:        35 | 37 | 39 | 41 | 43 | 45 | 47 | 49 | 51 | 53 => println!("there is no undefined behavior"),
tests/ui/collapsible_if.fixed:    // Test behavior wrt. `let_chains`.
tests/ui/collapsible_if.rs:    // Test behavior wrt. `let_chains`.
tests/ui/deprecated.stderr:error: lint `clippy::unsafe_vector_initialization` has been removed: the replacement suggested by this lint had substantially different behavior
tests/ui/mutex_atomic.stderr:error: consider using an `AtomicBool` instead of a `Mutex` here; if you just want the locking behavior and not the internal type, consider using `Mutex<()>`
tests/ui/mutex_atomic.stderr:error: consider using an `AtomicUsize` instead of a `Mutex` here; if you just want the locking behavior and not the internal type, consider using `Mutex<()>`
tests/ui/mutex_atomic.stderr:error: consider using an `AtomicIsize` instead of a `Mutex` here; if you just want the locking behavior and not the internal type, consider using `Mutex<()>`
tests/ui/mutex_atomic.stderr:error: consider using an `AtomicPtr` instead of a `Mutex` here; if you just want the locking behavior and not the internal type, consider using `Mutex<()>`
tests/ui/mutex_atomic.stderr:error: consider using an `AtomicPtr` instead of a `Mutex` here; if you just want the locking behavior and not the internal type, consider using `Mutex<()>`
tests/ui/mutex_atomic.stderr:error: consider using an `AtomicUsize` instead of a `Mutex` here; if you just want the locking behavior and not the internal type, consider using `Mutex<()>`
tests/ui/mutex_atomic.stderr:error: consider using an `AtomicIsize` instead of a `Mutex` here; if you just want the locking behavior and not the internal type, consider using `Mutex<()>`
tests/ui/redundant_clone.fixed:    s.clone().push_str("foo"); // OK, removing this `clone()` will change the behavior.
tests/ui/redundant_clone.rs:    s.clone().push_str("foo"); // OK, removing this `clone()` will change the behavior.
tests/ui/stable_sort_primitive.fixed:    // Negative examples: behavior changes if made unstable
tests/ui/stable_sort_primitive.rs:    // Negative examples: behavior changes if made unstable
tests/ui/uninit.stderr:error: this call for this type may be undefined behavior
tests/ui/uninit.stderr:error: this call for this type may be undefined behavior
tests/ui/uninit.stderr:error: this call for this type may be undefined behavior
```